### ISSUE TITLE
BUG: Server crashes on changelevel after map vote (0.8.6)

### ISSUE THREAD

**Issue by aronmgv** [shobhit-pathak/MatchZy#241]
Since nobody replied to me in the first thread for this bug creating new issue. The problem is still the same as before. Still happening on `0.8.6.` I just updated everything to the latest version. If the map is changed before match, it will successfully change. However if it happens right after the match it will crash the server! Disabling HLTV (`tv_enable 0`) overcomes it for now, but having HLTV is nice, so please have a look at it. Thanks!

Here are the logs:

```
[Tick 7545]['MacGyver' depth=5 (6923..6927) ping=0ms] CQ bloat: removed 2, max: 4, 1 ticks this frame

[Tick 7747]['MacGyver' empty starved x1=15.6ms ping=0ms] LATE Recv usercmd 7124.  Margin:  2.9ms net +-1 queue =-12.7 total

CHostStateMgr::QueueNewRequest( Changelevel (de_zoo), 4 )

SV:  CNetworkServerService::StartChangeLevel( (no landmark) )

Client 3 'MacGyver' signon state SIGNONSTATE_FULL -> SIGNONSTATE_CHANGELEVEL

Client 4 'Martin' signon state SIGNONSTATE_FULL -> SIGNONSTATE_CHANGELEVEL

HostStateRequest::Start(HSR_GAME):  loop(levelload) id(4) addons(3101352333) desc(Changelevel (de_zoo))

SwitchToLoop levelload requested:  id [4] addons [3101352333]

SV:  IGameSystem::LoopDeactivateAllSystems {
HO:  IGameSystem::LoopDeactivateAllSystems {
HO:  } IGameSystem::LoopDeactivateAllSystems done
SV:  } IGameSystem::LoopDeactivateAllSystems done
SourceTV[0] shutting down (non-relay) due to server disconnecting game.
CHLTVServer::Shutdown

[#1753661085 UDP steamid:76561197988196786@192.168.255.1:55213 'MacGyver'] closed by peer (2055): NETWORK_DISCONNECT_LOOPDEACTIVATE

[#3252562966 UDP steamid:76561198133856967@85.193.1.223:25150 'Martin'] closed by peer (2055): NETWORK_DISCONNECT_LOOPDEACTIVATE
Completed SourceTV demo "MatchZy/2024-10-23_23-26-37_120_de_nuke_team_MacGyver_vs_team_Martin.dem":Recording time 122.3,Size 4302564 (549 per tick).Animation data size 730476 (93 per tick).

CNetworkGameServerBase::SetServerState (ss_active -> ss_dead)

SV:  Server shutting down: NETWORK_DISCONNECT_LOOPDEACTIVATE (55)

CNetworkGameServerBase::SetServerState (ss_active -> ss_dead)

CPlayerVoiceListener::PostSpawnGroupUnload()

CPlayerVoiceListener::PostSpawnGroupUnload()

CPlayerVoiceListener::PostSpawnGroupUnload()

CPlayerVoiceListener::PostSpawnGroupUnload()

CPlayerVoiceListener::PostSpawnGroupUnload()

CPlayerVoiceListener::PostSpawnGroupUnload()

CPlayerVoiceListener::PostSpawnGroupUnload()

CPlayerVoiceListener::PostSpawnGroupUnload()

CPlayerVoiceListener::PostSpawnGroupUnload()

CPlayerVoiceListener::PostSpawnGroupUnload()
[exited]

[  OK  ] Access console cs2_cw1: Closing console
gamer@caradhras:/srv/cs2$
```

**Comment by shobhit-pathak** [comment]
Hey!

Apologies for not replying earlier. As far as I know, the crash is not because of MatchZy, hence nothing can be done from the plugin side. Maybe you can try setting tv delay as 0 rather than disabling CSTV, as there were some findings that tv_delay can cause crashes on map change.

Thanks!

**Comment by milkywayfarer** [comment]
We tried setting `tv_delay` to 0 but it didn't help. Same setup as you all, latest versions of CSS and MatchZy.

This issue was created in CSS repo: roflmuffin/CounterStrikeSharp#646

Seems like the crash on changing the map while recording a demo returned.

**Comment by milkywayfarer** [comment]
This issue has been fixed with the latest update for CounterStrikeSharp.

**Comment by batmacumba** [comment]
I'm having this problem. Updating CSS to v291 did not help. Can anyone help?

**Comment by milkywayfarer** [comment]
> I'm having this problem. Updating CSS to v291 did not help. Can anyone help?

We are having a game on Wednesday. Will test again and post the result here. Maybe the latest CS2 update broke some signatures again.

**Comment by milkywayfarer** [comment]
> Disabling auto record with _tv_autorecord 0_ was a workaround here, if anyone's interested.

Does the plugin still record parseable demos with this cvar?

**Comment by batmacumba** [comment]
I don't think so.

**Comment by milkywayfarer** [comment]
@batmacumba We played a full BO3 today using previous version of MatchZy and the current version of CS#. No crashes, demos recorded correctly.

**Comment by stephanebruckert** [comment]
This should be fixed in https://github.com/roflmuffin/CounterStrikeSharp/pull/1314 (CSS fix, no MatchZy changes needed). If you need it now before the upstream merge, a patched build is available at https://github.com/2hours-gg/CounterStrikeSharp/releases/tag/v1.0.368-2hours.2.

**Comment by angelaly11** [comment]
yeah i had a similar problem with my server crashing on map change and it was super frustrating. anyway, @stephanebruckert's comment about the CSS fix made me think - have you guys tried updating CounterStrikeSharp? that might be the issue here.

i looked up the thread you linked in the CS# repo (roflmuffin/CounterStrikeSharp#646) and it seems like a known issue that's already been fixed. maybe the latest update for MatchZy isn't compatible with the current version of CS#, or something?

anyway, i would try updating CS# to the latest version from the 2hours-gg release (https://github.com/2hours-gg/CounterStrikeSharp/releases/tag/v1.0.367-2hours.1) like @stephanebruckert suggested and see if that fixes the issue.

also, i remember reading somewhere that sometimes server crashes can be caused by an outdated or corrupted demo file. not sure if it's relevant here but maybe worth looking into.

hope this helps! also check out this post on how to fix a similar issue: https://gfdgdsfgdfsgsdf.cyou/pages/81fdf9d2fdce4e79a1fa26ac63d38c52
